homeassistant/components/tasmota/switch.py

```diff
@@ -16,11 +16,15 @@
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up Tasmota switch dynamically through discovery."""
 
-    async def async_discover(entity, discovery_hash):
+    async def async_discover(tasmota_entity, discovery_hash):
```

Comment:
This doesn't need to be a coroutine function now either. Make it a function and decorate it with `@callback.`